providers/implementations/encode_decode/decode_pem2der.c

```diff
@@ -183,6 +215,31 @@ static int pem2der_decode(void *vctx, OSSL_CORE_BIO *cin, int selection,
         char *data_type = (char *)pem_name_map[i].data_type;
         char *data_structure = (char *)pem_name_map[i].data_structure;
 
+        /*
+         * Since this may perform decryption, we need to check the selection to
+         * avoid password prompts for objects of no interest.
+         */
+        if (i <= PKCS8_LAST_IDX
+            && ((selection & OSSL_KEYMGMT_SELECT_PRIVATE_KEY)
+                || OPENSSL_strcasecmp(ctx->data_structure, "encryptedprivatekeyinfo") == 0
+                || OPENSSL_strcasecmp(ctx->data_structure, "privatekeyinfo") == 0)) {
+            ok = ossl_epki2pki_der_decode(der, der_len, selection, data_cb,
+                                          data_cbarg, pw_cb, pw_cbarg,
+                                          PROV_LIBCTX_OF(ctx->provctx),
+                                          ctx->propq);
+            goto end;
+        }
+
+        if (i <= SPKI_LAST_IDX
+            && ((selection & OSSL_KEYMGMT_SELECT_PUBLIC_KEY)
+                || OPENSSL_strcasecmp(ctx->data_structure, "subjectpublickeyinfo") == 0)) {
```

Comment:
Nit: make the capitalization similar to the other parts of the code?

Reply:
Done.